all: Use compile-time qstr when possible.

stinos opened the pull request:
For consistency qstr_from_str should be used for strings
known at runtime only.

dpgeorge commented on lib/utils/pyexec.c:
```diff
@@ -548,8 +548,8 @@ int pyexec_friendly_repl(void) {
-    mp_obj_t lcd_o = mp_call_function_0(mp_load_name(qstr_from_str("LCD")));
-    mp_call_function_1(mp_load_attr(lcd_o, qstr_from_str("light")), mp_const_true);
+    mp_obj_t lcd_o = mp_call_function_0(mp_load_name(MP_QSTR_LCD));
+    mp_call_function_1(mp_load_attr(lcd_o, MP_QSTR_light), mp_const_true);
```
I'd be happy to just get rid of this block of code.  It probably doesn't work anymore, and is highly specific to stm32.

stinos replied:
@dhylands I think you wrote this, any comments? Looking at other usaes of `USE_HOST_MODE` this does look deprecated indeed. Otherwsie it should rather be moved to a port-specific initialization?

dpgeorge replied:
I think I wrote this, see 75abee206d1a575aa98a486d043c94d64df432c1

stinos replied:
Right, didn't look further than pyexec's git history. Removed it, but you might want to adjust the commit message for clarity.

dpgeorge commented on ports/unix/main.c:
```diff
@@ -532,8 +532,8 @@ MP_NOINLINE int main_(int argc, char **argv) {
         MP_DECLARE_CONST_FUN_OBJ_0(extra_coverage_obj);
         MP_DECLARE_CONST_FUN_OBJ_0(extra_cpp_coverage_obj);
-        mp_store_global(QSTR_FROM_STR_STATIC("extra_coverage"), MP_OBJ_FROM_PTR(&extra_coverage_obj));
-        mp_store_global(QSTR_FROM_STR_STATIC("extra_cpp_coverage"), MP_OBJ_FROM_PTR(&extra_cpp_coverage_obj));
```
One could argue that `QSTR_FROM_STR_STATIC(...)` used here is a coverage test for that macro... because otherwise it's never used.  But maybe it's worth removing this macro altogether, since as you point out it should anyway be the compile-time value.

stinos replied:
Moreover it's actaully alomost the same as qstr_from_str (I doubt the compiler is going to generate different code).

dpgeorge replied:
I think the idea was that for static strings the strlen is known at compile time so can call `qstr_from_strn(str, len)` directly with a constant for the length.

stinos replied:
Probably, but I'm not sure how many compilers implement that. In any case I think removing is better for consistency and less confusing to have 2 ways to do almost the same thing.

stinos committed:
py/qstr.h: Remove QSTR_FROM_STR_STATIC macro.

It practically does the same as qtr_from_str and was only used in
one place, which should actually use the compile-time MP_QSTR_XXX
form for consistency; qstr_from_str is for runtime strings only.

stinos committed:
lib/utils/pyexec: Remove obsolete LCD initialization.

dpgeorge commented:
Merged in b9a35bebf75be53a817bf6341af14b882093e345 and fca2730ea01bc8457804884f4a0472ae0d938a19
